src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenFoodHandler.java

```diff
@@ -0,0 +1,144 @@
+import com.minecolonies.api.colony.interactionhandling.ChatPriority;
+import com.minecolonies.api.entity.citizen.citizenhandlers.ICitizenFoodHandler;
+import com.minecolonies.api.items.IMinecoloniesFoodItem;
+import com.minecolonies.core.colony.interactionhandling.StandardInteraction;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.StringTag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.Items;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashSet;
+import java.util.Queue;
+import java.util.Set;
+
+import static com.minecolonies.api.util.constant.Constants.TAG_STRING;
+import static com.minecolonies.api.util.constant.Constants.TICKS_SECOND;
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_LAST_FOODS;
+import static com.minecolonies.api.util.constant.TranslationConstants.*;
+
+/**
+ * The food handler for the citizen.
+ */
+public class CitizenFoodHandler implements ICitizenFoodHandler
+{
+    /**
+     * Assigned citizen data.
+     */
+    private final ICitizenData  citizenData;
+
+    /**
+     * Collection of last food items a citizen has eaten.
+     */
+    private final EvictingQueue<Item> lastEatenFoods = EvictingQueue.create(10);
+
+    /**
+     * Create the food handler.
+     * @param citizenData of it.
+     */
+    public CitizenFoodHandler(final ICitizenData citizenData)
+    {
+        super();
+        this.citizenData = citizenData;
+    }
+
+    @Override
+    public void addLastEaten(final Item item)
+    {
+        lastEatenFoods.add(item);
+        citizenData.markDirty(TICKS_SECOND);
+        if (lastEatenFoods.size() >= 10)
+        {
+            citizenData.triggerInteraction(new StandardInteraction(Component.translatable(NO + FOOD_DIVERSITY), ChatPriority.IMPORTANT));
+            citizenData.triggerInteraction(new StandardInteraction(Component.translatable(NO + FOOD_DIVERSITY), ChatPriority.IMPORTANT));
+            citizenData.triggerInteraction(new StandardInteraction(Component.translatable(NO + FOOD_QUALITY + URGENT), ChatPriority.BLOCKING));
+            citizenData.triggerInteraction(new StandardInteraction(Component.translatable(NO + FOOD_QUALITY + URGENT), ChatPriority.BLOCKING));
+        }
+    }
+
+    @Override
+    public Item getLastEaten()
+    {
+        return lastEatenFoods.peek();
+    }
+
+    @Override
+    public int checkLastEaten(final Item item)
+    {
+        int index = -1;
+        for (final Item foodItem : lastEatenFoods)
+        {
+            if (foodItem == item)
+            {
+                return index;
+            }
+            index++;
+        }
+        return -1;
+    }
+
+    @Override
+    public CitizenFoodStats getFoodHappinessStats()
+    {
+        int qualityFoodCounter = 0;
```

Comment:
should probably cache this until we ate another food, since its called regularly by the diseasemodifier